packages/custom_lint_builder/lib/src/analyzer_plugin/plugin_client.dart

```diff
@@ -512,22 +516,13 @@ abstract class ClientPlugin {
   /// list of [excluded] resources to the given [driver].
   void _addFilesToDriver(
     AnalysisDriver driver,
-    Resource resource,
-    List<String> excluded,
+    List<String> include,
   ) {
-    final path = resource.path;
-    if (excluded.contains(path)) return;
-    if (resource is File && extension(path) == '.dart') {
-      driver
-        ..addFile(path)
-        ..getResult(path);
-    } else if (resource is Folder) {
-      try {
-        for (final child in resource.getChildren()) {
-          _addFilesToDriver(driver, child, excluded);
-        }
-      } on FileSystemException {
-        // The folder does not exist, so ignore it.
+    for (final path in include) {
```

Comment:
printing the include list here (filtering to paths that end in .dart) I only see main.dart. So I don't know where it is getting main.g.dart

Reply:
Probably on priority file change. 

I believe files outside exclude/include are still passed to the event. THis could cause opening main.g.dart to analyze the file. Then custom_lint would report lints on it